Utilities/bootstrap

```diff
@@ -229,34 +231,15 @@ def get_ninja_path(args):
 
 def get_build_target(args):
     """Returns the target-triple of the current machine."""
-    if platform.system() == 'Darwin':
-        return "x86_64-apple-macosx"
-    elif platform.system() == 'Linux':
-        if 'ANDROID_DATA' in os.environ:
-            if platform.machine().startswith('armv7'):
-                return 'armv7a-unknown-linux-androideabi'
-            elif platform.machine() == 'aarch64':
-                return 'aarch64-unknown-linux-android'
-            else:
-                raise SystemExit("ERROR: unsupported Android platform:", platform.machine())
-        elif platform.machine() == 'x86_64':
-            return "x86_64-unknown-linux-gnu"
-        elif platform.machine() == "i686":
-            return "i686-unknown-linux"
-        elif platform.machine() == 's390x':
-            return "s390x-unknown-linux"
-        elif platform.machine() == 'ppc64le':
-            return 'powerpc64le-unknown-linux'
-        elif platform.machine().startswith("armv7"):
-            return 'armv7-unknown-linux-gnueabihf'
-        elif platform.machine() == 'aarch64':
-            return 'aarch64-unknown-linux'
+    try:
+        target_info_json = subprocess.check_output([args.swiftc_path, '-print-target-info'], universal_newlines=True).strip()
+    except Exception as e:
+        if platform.system() == 'Darwin':
+            return 'x86_64-apple-macosx'
         else:
-            raise SystemExit("ERROR: unsupported machine:", platform.machine())
-    elif platform.system() == 'Windows':
-        return 'x86_64-unknown-windows-msvc'
-    else:
-        raise SystemExit("ERROR: unsupported system:", platform.system(), platform.machine())
+            error(str(e))
+    args.target_info = json.loads(target_info_json)
+    return args.target_info["target"]["triple"]
```

Comment:
It seems like we get the triple as "x86_64-apple-macosx10.15" on Darwin. @DougGregor is adding a `unversionedTriple` field which we can use instead.

Reply:
@buttaface but you can also take that up if you're interested in making a patch to the Swift compiler 😅

Reply:
[There is a `moduleTriple` already](https://github.com/apple/swift/commit/61c83d241#diff-17ea82c601ab471fe0c3cdd9bc000016R1880), maybe I can just use that? I didn't know what it was before, but experimenting with it a bit now, it seems to remove the OS version for both OS X and Android, returning `x86_64-apple-macos` for macOS 10.14.

Reply:
We need it to say "macosx" for the third component. Module triple seems to say "macos" 🙁

Reply:
Not sure why this matters? This is only used to build SwiftPM, not by SwiftPM to build Swift packages.

Reply:
Oh, the build dir from the first bootstrap step?

Reply:
Bootstrap is the first step. We want to use this when building Swift packages as well so we avoid hardcoding the triples in SwiftPM codebase.

Reply:
Alright, but for the purpose of this pull, ie simply building the SPM binaries, it seems that both the currently provided `triple` and `moduleTriple` should work fine? Unless there's some intricacy of swiftmodules or something else I'm unaware of.

Reply:
Some of the bootstrap commands are expecting the build directory from the bootstrap script and swiftpm to match so they'll fail on Darwin. The Darwin smoke test bot should surface those errors.

Reply:
Hmm, looks like it built fine, only for `swift-test` to finally fail.

Reply:
Yeah, I think because it ends up looking in the wrong directory due to the version number in the triple.

Reply:
`unversionedTriple` PR is here: https://github.com/apple/swift/pull/28874

Reply:
... and now it's merged to master, which 5.2 coming along soon https://github.com/apple/swift/pull/28875

Reply:
... and also merged to 5.2.